scripts/init.d/cgconfig.in

```diff
@@ -30,37 +30,54 @@
 ### END INIT INFO
 
 # get correct location of binaries from configure
-prefix=@prefix@;exec_prefix=@exec_prefix@;sbindir=@sbindir@
+sbindir=@sbindir@
 CGCONFIGPARSER_BIN=$sbindir/cgconfigparser
 CONFIG_FILE=/etc/cgconfig.conf
+CONFIG_DIR=/etc/cgconfig.d
 servicename=cgconfig
-lockfile=/var/lock/subsys/$servicename
 
+
+lockfile=/run/subsys/$servicename
```

Comment:
@lockywolf Thanks for working on the script.  The patch looks good, just a few minor nitpicks. Should the `lockfile`  be `/run/lock/subsys/$servicename`?

Reply:
I am not very sure. FHS is speaking about PID files (https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s13.html), and file locks (https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s09.html).

But the libcgroup lock is neither of those, it is saying that "the subsys has been initialized", nothing more.

On Slackware, all of `/var/lock`, `/run`, and `/run/lock`, have something resembling lock files.
Do you insist on `/run/lock/subsys/$servicename`.

Reply:
Following the naming convention from the `sysv` world,  where it's expected to have the name as `/var/lock/subsys/$servicename`, for it to work without any issues and now that  `/var/lock/` got moved to  `/run/lock/`, let's retain it.  @drakenclimber what are your thoughts on this.

Reply:
(Not an expert here.)

On my Ubuntu 20 box, I don't have a `/run/subsys` dir, but I do have a `/run/lock/subsys` dir.  My Fedora 35 box matches the Ubuntu one.

Looking through systemd's source, it looks like `/run/lock/subsys/` is the way forward:
https://github.com/systemd/systemd/blob/v250/tmpfiles.d/legacy.conf.in#L22

Reply:
> But the libcgroup lock is neither of those, it is saying that "the subsys has been initialized", nothing more.
> 
> On Slackware, all of `/var/lock`, `/run`, and `/run/lock`, have something resembling lock files. Do you insist on `/run/lock/subsys/$servicename`.

At the moment, it is rather simple.  But what if it grows into something more powerful?  What's the harm in treating like a more advanced lock?

Reply:
One more foolish question while I am at it - should we consider a complete re-write of this file?  Looking through its history, it looks like it started out from a generic template and organically grew to what it is today.  (Nothing wrong with that.)  But the world has changed greatly since then.  Perhaps systemd has a better template/recommendation that what we currently have?

Reply:
I have moved the lock to /run/lock/subsys/.

I have no opposition to systemd, but I think that this script is now aimed at non-systemd systems. After all, systemd has its own cgroup management facility, whereas libcgroup can be used at a far greater variety of setups, not necessarity GNU. (Which is also why I have added not just pre-LSB compatibility logging, but also bare "printf" fallbacks.)

Having a separate systemd unit would probably be the way to go with systemd systems.